test/Microsoft.AspNetCore.Razor.Language.Test/TestFiles/ParserTests/TagHelperParseTreeRewriterTest/AllowsPrefixedTagHelpers10.stree.txt

```diff
@@ -1,13 +1,35 @@
-Markup block - Gen<None> - 47 - (0:0,0)
-    Tag block - Gen<TagHelper> - 47 - (0:0,0) - th:myth - mythTagHelper
-        StartTagAndEndTag - <th:myth class="btn"> ... </th:myth>
-        class - DoubleQuotes
-            Markup span - Gen<Markup> - [btn] - SpanEditHandler;Accepts:Any - (16:0,16) - Tokens:1
-                SyntaxKind.Text;[btn];
-        SyntaxKind.HtmlTextLiteral - [words and spaces] - [21..37) - FullWidth: 16 - Slots: 1
-            SyntaxKind.List - [words and spaces] - [21..37) - FullWidth: 16 - Slots: 5
-                SyntaxKind.Text;[words];
-                SyntaxKind.Whitespace;[ ];
-                SyntaxKind.Text;[and];
-                SyntaxKind.Whitespace;[ ];
-                SyntaxKind.Text;[spaces];
+RazorDocument - [0..47)::47 - [<th:myth class="btn">words and spaces</th:myth>]
+    MarkupBlock - [0..47)::47
+        MarkupTagHelperElement - [0..47)::47 - th:myth[StartTagAndEndTag] - mythTagHelper
+            MarkupTagHelperStartTag - [0..21)::21
+                MarkupTextLiteral - [0..8)::8 - [<th:myth] - Gen<Markup> - SpanEditHandler;Accepts:Any
+                    OpenAngle;[<];
+                    Text;[th:myth];
+                MarkupTagHelperAttribute - [8..20)::12 - class - DoubleQuotes
+                    MarkupTextLiteral - [8..9)::1 - [ ] - Gen<Markup> - SpanEditHandler;Accepts:Any
+                        Whitespace;[ ];
+                    MarkupTextLiteral - [9..14)::5 - [class] - Gen<Markup> - SpanEditHandler;Accepts:Any
+                        Text;[class];
+                    Equals;[=];
+                    MarkupTextLiteral - [15..16)::1 - ["] - Gen<None> - SpanEditHandler;Accepts:Any
+                        DoubleQuote;["];
+                    MarkupTagHelperAttributeValue - [16..19)::3
+                        MarkupLiteralAttributeValue - [16..19)::3 - [btn]
+                            MarkupTextLiteral - [16..19)::3 - [btn] - Gen<Markup> - SpanEditHandler;Accepts:Any
+                                Text;[btn];
+                    MarkupTextLiteral - [19..20)::1 - ["] - Gen<None> - SpanEditHandler;Accepts:Any
+                        DoubleQuote;["];
+                MarkupTextLiteral - [20..21)::1 - [>] - Gen<Markup> - SpanEditHandler;Accepts:Any
+                    CloseAngle;[>];
+            MarkupTextLiteral - [21..37)::16 - [words and spaces] - Gen<Markup> - SpanEditHandler;Accepts:Any
+                Text;[words];
+                Whitespace;[ ];
+                Text;[and];
+                Whitespace;[ ];
+                Text;[spaces];
+            MarkupTagHelperEndTag - [37..47)::10
```

Comment:
This is probably the main structural change. Tag helper end tags are now well represented in the tree.
I am currently not generating the corresponding classified spans for this to keep them consistent with the old tree. But ideally we should generate those. We should discuss this at some point. I've added it to the tracking list https://github.com/aspnet/Razor/issues/2619. @rynowak this is what I was trying to explain earlier.

Reply:
If the behaviour isn't changing right now, then that sounds great!